src/main/java/edu/harvard/iq/dataverse/FileDownloadHelper.java

```diff
@@ -36,16 +46,229 @@
+        String questionNo = (String) ((UIOutput) e.getSource()).getId();
+         String position= (String) ((UIOutput) e.getSource()).getValue();
+     }
+    
     public FileDownloadHelper() {
         this.filesForRequestAccess = new ArrayList<>();
     }
 
     
+     private boolean testResponseLength(String value) {
+        return !(value != null && value.length() > 255);
+     }
+     
+     private boolean validateGuestbookResponse(GuestbookResponse guestbookResponse){
+                Dataset dataset = guestbookResponse.getDataset();
+                boolean valid = true;
+         if (dataset.getGuestbook() != null) {
+             if (dataset.getGuestbook().isNameRequired()) {
+                 boolean nameValid = (guestbookResponse.getName() != null && !guestbookResponse.getName().isEmpty());
+                 if (!nameValid) {
+                     nameField.setValid(false);
+                     FacesContext.getCurrentInstance().addMessage(nameField.getClientId(),
+                             new FacesMessage(FacesMessage.SEVERITY_ERROR, BundleUtil.getStringFromBundle("requiredField"), null));
+                 }
+                 valid &= nameValid;
+             }
+                valid &= testResponseLength(guestbookResponse.getName());
+                 if (! testResponseLength(guestbookResponse.getName())){
+                    nameField.setValid(false);
+                     FacesContext.getCurrentInstance().addMessage(nameField.getClientId(),
+                             new FacesMessage(FacesMessage.SEVERITY_ERROR, BundleUtil.getStringFromBundle("dataset.guestbookResponse.guestbook.responseTooLong"), null));
+                 }
+             if (dataset.getGuestbook().isEmailRequired()) {
+                 boolean emailValid = (guestbookResponse.getEmail() != null && !guestbookResponse.getEmail().isEmpty());
+                 if (!emailValid) {
+                     emailField.setValid(false);
+                     FacesContext.getCurrentInstance().addMessage(emailField.getClientId(),
+                             new FacesMessage(FacesMessage.SEVERITY_ERROR, BundleUtil.getStringFromBundle("requiredField"), null));
+                 }
+                 valid &= emailValid;
+             }
+                valid &= testResponseLength(guestbookResponse.getEmail());
+                 if (! testResponseLength(guestbookResponse.getEmail())){
+                    emailField.setValid(false);
+                     FacesContext.getCurrentInstance().addMessage(emailField.getClientId(),
+                             new FacesMessage(FacesMessage.SEVERITY_ERROR, BundleUtil.getStringFromBundle("dataset.guestbookResponse.guestbook.responseTooLong"), null));
+                 }
+            if (dataset.getGuestbook().isInstitutionRequired()) {
+                 boolean institutionValid = (guestbookResponse.getEmail() != null && !guestbookResponse.getEmail().isEmpty());
```

Comment:
This seems to be a copy/paste error. It should be the institution rather than the email. I found this bug by accident when I saw that I can't submit the form even when supplying an institution when required:

![screen shot 2018-01-11 at 12 27 33 pm](https://user-images.githubusercontent.com/21006/34838143-06a2c892-f6cb-11e7-9a2f-3a816863e6c7.png)